docs/madr/decisions/019-offline-signing-tokens.md

```diff
@@ -0,0 +1,207 @@
+# Offline signing tokens
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4031
+
+## Context and Problem Statement
+
+Kuma provides multiple tokens for different lines of security
+* User authenticating to Kuma API Server
+* Data plane proxy authenticating to Kuma DP Server
+* Zone proxy authenticating to Kuma DP Server
+
+Right now, a user has to use Kuma CP to issue the tokens.
+While this works, some users may prefer using their existing token issuer.
+We can provide an interface of a token (document required fields) and a way to provide public key of a token.
+This way a user can use different system to issue token, but let Kuma CP validate the token.
+This improves security posture if you already have a safe way of managing signing key for your tokens.
+
+## Considered Options
+
+* CP configuration and kumactl improvements
+
+## Decision Outcome
+
+Chosen option: "CP configuration and kumactl improvements".
+
+## Pros and Cons of the Options
+
+### User token configuration
+
+```yaml
+apiServer:
+  authn:
+    type: tokens
+    tokens:
+      bootstrapAdminToken: true
+      enableIssuer: true
+      validator:
+        useSecrets: true
+        publicKeys:
+        - kid: 123
+          key: |
+            -----BEGIN RSA PUBLIC KEY-----
+            MIIEogIBAAKCAQEAy0KtfI7O0TJ00............
+            -----END RSA PUBLIC KEY-----
```

Comment:
We need to allow this to be a file for convenience.

Reply:
added